### PR TITLE
Setup and Write E2E tests

### LINKED ISSUE #55: Write Cypress E2E test scripts
### Is your enhancement related to a problem? Please describe.

Write Cypress E2E test scripts for automated testing

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mehul0810**
### Description of the Change

Closes #55

### Changelog Entry

> Added - New feature

### Credits

Props @mehul0810

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        key: ${{ hashFiles('package-lock.json') }}
+
+    - name: Install dependencies
+      run: npm install
+
+    - name: Composer install
+      run: composer install
+
+    - name: Build
+      run: npm run build
```
Comment (iamdharmesh) [inline .github/workflows/cypress.yml:85]: As we don't have a `build` script in package.json, we need to remove this.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		"WordPress-Plugin",
+		"New Relic"
+	],
+	"homepage": "https://github.com/10up/wp-newrelic#readme",
+	"repository": "git+https://github.com/10up/wp-newrelic.git",
+	"bugs": {
+		"url": "https://github.com/10up/wp-newrelic/issues"
+	},
+	"engines": {
+		"node": ">=12.13.0"
```
Comment (iamdharmesh) [inline package.json:22]: Maybe update this to v16 to match what we have in `.nvmrc`.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		"distributor.php",
+		".github/workflows/*",
+		".gitattributes",
+		"assets/img/*",
+		"dist/**/*",
+		"includes/**/*",
+		"lang/**/*",
+		"templates/**/*",
+		"vendor/yahnis-elsts/plugin-update-checker/**/*"
+	]
```
Comment (iamdharmesh) [inline package.json:56]: Noticed some files which are not related to this repo. eg: "distributor.php", "vendor/yahnis-elsts/plugin-update-checker/**/*" etc... Could you please remove those from the list? Thanks

File: tests/Basic-Auth-master/basic-auth.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<?php
```
Comment (iamdharmesh) [inline tests/Basic-Auth-master/basic-auth.php:1]: I'm not sure if we need this plugin in the test environment, as I don't see its usage anywhere in tests. Could you please remove it if are not using it? Thanks

File: tests/cypress/reports/mochawesome.json [hunk 5]
```diff
@@ -0,0 +1,110 @@
+{
```
Comment (iamdharmesh) [inline tests/cypress/reports/mochawesome.json:1]: It looks like this report file is accidentally committed to the repo, Could you Please add the `tests/cypress/reports` folder to `.gitignore`? It contains test report files and shouldn't be pushed to the repo.

File: tests/bin/initialize.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+npm run env run tests-cli "wp rewrite structure '/%postname%/' --hard"
+
+status=0
+npm run env run tests-cli "wp site list" || status=$?
+
+if [ $status -eq 0 ]
+then
+	echo "Multisite already initialized"
+else
+	echo "Converting to multisite"
```
Comment (iamdharmesh) [inline tests/bin/initialize.sh:14]: Do we need a multisite environment here? Not very much familiar with the plugin but based on functionality it seems we don't require a multisite test setup here. Please let me know if I am missing anything here. Thanks

**Review by iamdharmesh — CHANGES_REQUESTED** [review]
Hi @mehul0810, 

Thanks a lot for working on this and raising the PR. Added some comments/notes on PR to check and discuss could you please help to check on that once? Please feel free to reach out if you need any further information on comments.

Thank you.

File: tests/bin/initialize.sh (hunk 6 above)
Comment (jeffpaul) [reply tests/bin/initialize.sh:14]: @iamdharmesh might be best to have you work on updates here and merge this in as you're able

**Comment by iamdharmesh** [comment]
Closing in favour of #83

File: tests/bin/initialize.sh (hunk 6 above)
Comment (iamdharmesh) [reply tests/bin/initialize.sh:14]: @jeffpaul raised new PR for this #83
